### PR TITLE
Feat/post

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. User can add a post by on click of 'Add Post' button on left menu bar
2. tailwind has been upgraded to v3

### CODE REVIEW

**Review by sashaboi — APPROVED** [review]
Looks great, and functionality works as expected, good to merge ! 

A small suggestion : the add post button is firing the created post and the error toast both one after the other, I hope you will handle that in coming versions!

File: src/PostModalPortal.jsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import ReactDOM from "react-dom";
+import {PostModal} from 'features';
+
+const PostModalPortal = () => {
+  return ReactDOM.createPortal(<PostModal/>, document.getElementById('post-modal'));
+}
+
+export {PostModalPortal};
```
Comment (bharati-21) [inline src/PostModalPortal.jsx:8]: Nice to see you have used Portals!

File: src/custom-hooks/useClickOutside.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      return () => {
+        document.removeEventListener("mousedown", listener);
+        document.removeEventListener("touchstart", listener);
+      };
+    },
+    [ref, handler]
+  )
+}
+
+export { useClickOutside }
```
Comment (bharati-21) [inline src/custom-hooks/useClickOutside.jsx:24]: Nice way of using hooks to create outside click feature for modals!

File: src/features/post/postSlice.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      state.posts = payload;
+    })
+    builder.addCase(addPost.rejected, (state) => {
+      state.postLoading = false;
+      state.postError = 'Error in adding post';
+    })
+  }
+})
+
+// export const { } = postSlice.actions;
```
Comment (bharati-21) [inline src/features/post/postSlice.js:71]: You can remove commented code.

File: src/service/post.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+
+const getAllPostService = async () => {
+  const token = localStorage.getItem('token');
+  console.log('from get all posts service');
```
Comment (bharati-21) [inline src/service/post.js:5]: You can remove the logs

File: src/views/home/index.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  useEffect(() => {
+    (async () => {
+      try{
+       const response = await dispatch(getAllPosts());
+        if(response?.error)
+          throw new Error('Error in loading posst');
+      }
+      catch(error){
+        console.log(error.message);
```
Comment (bharati-21) [inline src/views/home/index.jsx:20]: You can remove the logs.

**Review by bharati-21 — COMMENTED** [review]
Great to see the extensive use of tailwind utilities. Also nice to see you have used the power of @utilties base, components! Easy to understand code, and good folder structure!

Looks good to merge.

**Comment by SHIKHASINGH1506** [comment]
Thank you for the review

**Pull request merged**